rules/rules-reviewed/azure/azure-message-queue.windup.xml

```diff
@@ -184,6 +184,34 @@
                 <technology-tag level="INFORMATIONAL">AMQP Client</technology-tag>
             </perform>
         </rule>
+        <rule id="azure-message-queue-spring-jms-rabbitmq-01000">
+            <when>
+                <and>
+                    <project>
+                        <artifact groupId="org.springframework{*}" artifactId="{*}jms"/>
+                    </project>
+                    <project>
+                        <artifact groupId="com.rabbitmq{*}" artifactId="rabbitmq-jms"/>
+                    </project>
+                </and>
+            </when>
+            <perform>
+                <hint title="Spring JMS RabbitMQ dependency found" category-id="information" effort="0">
+                     <message>
+                        The application uses Spring JMS RabbitMQ dependency. It is recommended to migrate to Azure Service Bus with Spring JMS
+
+                        * Replace RabbitMQ dependencies with Azure Service Bus dependencies: Remove RabbitMQ dependencies and add Azure Service Bus dependencies: com.azure.spring spring-cloud-azure-dependencies, com.azure.spring spring-cloud-azure-starter-servicebus-jms.
+                        * Replace RabbitMQ connection details with Azure Service Bus connection details: Modify the Java application's configuration updating the connection information and message queues/topics.
+                    </message>
+                    <link title="Use JMS in Spring to access Azure Service Bus" href="https://learn.microsoft.com/azure/developer/java/spring-framework/configure-spring-boot-starter-java-app-with-azure-service-bus"/>
+                    <link title="Use Azure Service Bus with JMS" href="https://learn.microsoft.com/azure/developer/java/spring-framework/spring-jms-support"/>
+                    <link title="Azure Service Bus documentation" href="https://learn.microsoft.com/azure/service-bus-messaging"/>
+                    <link title="Use Azure Service Bus in Spring applications" href="https://learn.microsoft.com/azure/developer/java/spring-framework/using-service-bus-in-spring-applications"/>
+                    <tag>RabbitMQ Client</tag>
+                </hint>
+                <technology-tag level="INFORMATIONAL">RabbitMQ Client</technology-tag>
```

Comment:
@agoncal Made them OPTIONAL

Reply:
@agoncal I made the category-id of the hints "optional" based on your suggestion. I didn't change the technology-tag level because it includes only IMPORTANT and INFORMATIONAL. See the definition here: https://github.com/windup/windup/blob/master/reporting/api/src/main/java/org/jboss/windup/reporting/model/TechnologyTagLevel.java